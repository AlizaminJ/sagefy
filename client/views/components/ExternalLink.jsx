import React from 'react'
import { node } from 'prop-types'

export default function ExternalLink(props) {
  const { children } = props
  return (
    <a {...props} target="_blank" rel="noopener noreferrer">
      {children}
    </a>
  )
}

ExternalLink.propTypes = {
  children: node.isRequired,
}
