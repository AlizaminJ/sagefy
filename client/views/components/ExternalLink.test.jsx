import React from 'react'
import ExternalLink from './ExternalLink'

describe('ExternalLink component', () => {
  it('should render an external link', () => {
    expect(
      ExternalLink({
        href: 'https://example.com',
        children: <span>Hello!</span>,
      })
    ).toMatchSnapshot()
  })
})
