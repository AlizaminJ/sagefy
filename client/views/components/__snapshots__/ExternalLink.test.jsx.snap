// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink component should render an external link 1`] = `
<a
  href="https://example.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <span>
    Hello!
  </span>
</a>
`;
